### ISSUE TITLE
Method "__getitem__" cannot be called on Sequence

### ISSUE THREAD

**Issue by AlexanderPodorov** [microsoft/pyright#7239]
**Describe the bug**
See example code snippet

```python
from collections.abc import Sequence
from dataclasses import dataclass
from typing import Any, Generic, TypeVar, cast

from pydantic import TypeAdapter

# 1. use case from pydantic

v = TypeAdapter(Sequence[int]).validate_python([1, 2, 3])

v[0]  # shows warning


# 2. MRE based on the use case from pydantic

T = TypeVar('T')


@dataclass
class Test(Generic[T]):
    t: type[T]

    def test(self) -> T:
        # pretend the instance of T has been built in some way
        return cast(Any, [])


g = Test(Sequence[int]).test()

g[0]  # shows warning


# 3. Adjusted example returning generic sequence

@dataclass
class TestSeq(Generic[T]):
    t: type[T]

    def test(self) -> Sequence[T]:
        return []


h = TestSeq(int).test()

h[0]  # works fine, no warnings

```

`pydantic==2.6.1`

**VS Code extension or command-line**
Pylance v2024.2.1 (Pyright 1.1.348)

**Comment by erictraut** [comment]
This issue has been fixed in later versions of pyright. Please update to pyright 1.1.350 or to the prerelease version of pylance.
